src/HEVM.hs

```diff
@@ -1,446 +1,365 @@
-{-# Language RecordWildCards #-}
-{-# Language TypeApplications #-}
-{-# Language OverloadedStrings #-}
-{-# Language DataKinds #-}
-{-# Language GADTs #-}
-{-# Language MonadComprehensions #-}
-{-# Language ViewPatterns #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module HEVM where
 
-import Prelude hiding (lookup, GT, LT)
-
-import Syntax
-import Syntax.Annotated as Annotated
-
-import Data.ByteString.UTF8 (toString)
-import Data.Text (Text, pack, splitOn)
-import Data.Maybe
-import Data.List hiding (lookup)
-import Data.Map hiding (drop, null, findIndex, splitAt, foldl, foldr, filter)
-import qualified Data.Map as Map
-import Control.Monad.State.Strict (execState)
-import Data.SBV.Control
-import Data.SBV
-import Data.SBV.String ((.++), subStr)
-import Control.Lens hiding (op, pre, (.>))
+import qualified Data.Map as M
+import Data.List
+import qualified Data.Text as T
+import qualified Data.Text.Lazy.IO as TL
+import qualified Data.ByteString.Char8 as B8 (pack)
+import Control.Concurrent.Async
 import Control.Monad
-import qualified Data.Vector as Vec
+import Data.DoubleWord
 
-import Print
+import Syntax.Annotated
```

Comment:
```suggestion
import qualified Syntax.Annotated as Act
```

Reply:
Nice idea! What do you think about HEVM.Types? We could leave it unqualified since there are no name collisions anymore. Or maybe a better qualifier like HEVM? Right now the qualifier is `Types` and I don't like it at all.

Reply:
I think it is a nice idea! It will force us to write a lot of `Act.`s in the translation, but is more consistent with the way we import HEVM's ast. Also not sure..